astropy/wcs/wcs.py

```diff
@@ -636,7 +637,8 @@ def calc_footprint(self, header=None, undistort=True, axes=None, center=True):
                 naxis2 = header.get('NAXIS2', None)
 
         if naxis1 is None or naxis2 is None:
-            return None
+            raise ValueError(
+                    "image size could not be determined.")
```

Comment:
Maybe a better message would specify that either `axes` or `header` should be specified (but not both).  Another problem with this function is that if `axes` is not a two-element object this will fail above on [line 624](https://github.com/astropy/astropy/pull/3557/files#diff-d4c5b9e74c6c26000684075497033504R624) with a traceback.  So as long as we're validating inputs we might as well catch that possibility too.  I agree though that if neither is provided it makes more sense to raise an exception than to just return `None`.  I think a `TypeError` would be more appropriate, but I think that's also open to interpretation.  @astrofrog?